cluster-autoscaler/ca.patch

```diff
@@ -1,3 +1,103 @@
+diff --git a/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go b/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go
+index 7fb533570..18e8adca9 100644
+--- a/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go
++++ b/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go
+@@ -604,7 +604,10 @@ func (o *ScaleUpOrchestrator) SchedulablePodGroups(
+ 	var schedulablePodGroups []estimator.PodEquivalenceGroup
+ 	for _, eg := range podEquivalenceGroups {
+ 		samplePod := eg.Pods[0]
+-		if err := o.autoscalingContext.PredicateChecker.CheckPredicates(o.autoscalingContext.ClusterSnapshot, samplePod, nodeInfo.Node().Name); err == nil {
++		err := o.autoscalingContext.PredicateChecker.CheckPredicates(o.autoscalingContext.ClusterSnapshot, samplePod, nodeInfo.Node().Name)
++		// Ignore inter-pod affinity since this is not usually a reason to fail the ASG
++		// (unless the anti-affinity conflict is with a DaemonSet pod, but there's no way to tell)
++		if err == nil || err.PredicateName() == "InterPodAffinity" {
+ 			// Add pods to option.
+ 			schedulablePodGroups = append(schedulablePodGroups, estimator.PodEquivalenceGroup{
+ 				Pods: eg.Pods,
+diff --git a/cluster-autoscaler/estimator/binpacking_estimator.go b/cluster-autoscaler/estimator/binpacking_estimator.go
+index 229a2e264..f145a218f 100644
+--- a/cluster-autoscaler/estimator/binpacking_estimator.go
++++ b/cluster-autoscaler/estimator/binpacking_estimator.go
+@@ -108,13 +108,18 @@ func (e *BinpackingNodeEstimator) Estimate(
+ 		var err error
+ 		var remainingPods []*apiv1.Pod
+ 
++		klog.V(4).Infof("trying to schedule %d pods on existing nodes", len(podsEquivalenceGroup.Pods))
```

Comment:
Let's remove the print debugging?

Reply:
Honestly, I'd prefer to keep it in unless you strongly think we should take it out. If we run into issue with this again it'd be a lot easier to debug with the debugging messages than of course, adding them back and the back and forth that entails. Though - we default to v=4 I believe so I'd be happy to give it a higher verbosity (5?)

Reply:
In general, I'd like to minimize the size of our patch, to avoid conflicts when updating.

So in this case, if you think the debugging is useful, let's keep it in but revisit in 1-3 months to see if we can get rid of it?